### PR TITLE
fix(release_health): Improve behavior around empty-string queries [INGEST-1193]

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: src/sentry/snuba/metrics/query_builder.py [hunk 1]
```diff
@@ -119,6 +119,12 @@ def resolve_tags(org_id: int, input_: Any) -> Any:
         if input_.function == "ifNull":
             # This was wrapped automatically by QueryBuilder, remove wrapper
             return resolve_tags(org_id, input_.parameters[0])
+        elif input_.function == "isNull":
+            return Condition(
+                lhs=resolve_tags(org_id, input_.parameters[0]),
+                op=Op.EQ,
+                rhs=resolve_tags(org_id, ""),
+            )
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/query_builder.py:127]: Does this also work for nested conditions like `release:foo or release:""`. I think that might resolve to `Function("or", [Function("equals", ...), ..]` and then you get a `Condition` as a function argument.

File: tests/snuba/api/endpoints/test_organization_sessions.py [hunk 2]
```diff
@@ -1350,3 +1352,31 @@ def test_unrestricted_date_range(self):
...
+            )
+        )
+        # Empty strings are invalid values for releases and environments, but we should still handle those cases correctly at the query layer
+        response = self.do_request(
+            {
+                "project": self.project.id,  # project without users
+                "statsPeriod": "1d",
+                "interval": "1d",
+                "field": ["sum(session)"],
+                "query": 'release:"" environment:""',
```
Comment (jjbayer) [inline tests/snuba/api/endpoints/test_organization_sessions.py:1370]: `release` is required and `environment` is optional, right? So should we really treat them the same? Or should we resolve `environment:""` to "has no environment tag"?

Comment (untitaker) [reply tests/snuba/api/endpoints/test_organization_sessions.py:1370]: I don't think we should treat them differently. "has no tag" is typically expressed in discover as `!tag:*`

File: src/sentry/snuba/metrics/query_builder.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/snuba/metrics/query_builder.py:129]: Thanks, fixed. I believe we should essentially never convert Function to Condition or vice-versa in resolve_tags (though we already do)

File: src/sentry/utils/snuba.py [hunk 3]
```diff
@@ -138,6 +138,7 @@
     "<": "less",
     ">=": "greaterOrEquals",
     "<=": "lessOrEquals",
+    "IS NULL": "isNull",
```
Comment (jjbayer) [inline src/sentry/utils/snuba.py:141]: Why is this addition necessary?

File: src/sentry/snuba/metrics/query_builder.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/snuba/metrics/query_builder.py:129]: I agree, what other conversion instance is there? I can't seem to spot it.

**Review by jjbayer — APPROVED** [review]

File: src/sentry/snuba/metrics/query_builder.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/snuba/metrics/query_builder.py:129]: potentially line 121, though i believe conversion between "expressions" and functions might be fine

File: src/sentry/utils/snuba.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/utils/snuba.py:141]: otherwise the test crashes in discover code before we get to resolve_tags. I am not sure why but the mapping seems to make sense to me
